src/rickbox/database/models/user.py

```diff
@@ -0,0 +1,141 @@
+"""
+Object representation for the users table / user entities.
+Acts as the interface between the database and the application.
+
+Usage:
+
+The constructor takes in one required param – username – and two optional
+params – name, and email. Returns a User object with the given properties.
+"""
+
+import json
+
+from database.database_helper import DatabaseHelper
+
+
+class User:
+    id = None
+    username = None
+    name = None
+    email = None
+
+    def __init__(self, id, username, name, email):
+        self.id = id
+        self.username = username
+        self.name = name
+        self.email = email
+
+    def save(self):
+        """
+        Commits a User record to the database.
+
+        Returns:
+            If the commit was successful, return the ID of the newly-created User
+
+        Raises:
+            ValueError:
+                - user.id is set (`put` operation is NYI)
+                - user.username is blank (None or '')
+            LookupError:
+                - User record could not be found after saving
+            NameError:
+                - user.username exists in database already
+        """
+        if self.id is not None:
+            # currently only supports saving new records, not updating existing ones
+            raise ValueError('The `User.id` attribute must be None to save a User object.')
+
+        if self.username in (None, ''):
+            raise ValueError('The `User.username` attribute must not be empty to save a User object.')
+
+        username_taken = True
+        try:
+            User.get_by_username(self.username)
+        except LookupError:
+            # Username is not claimed. Proceed with creation
+            username_taken = False
+
+        if username_taken:
+            # This should probably be some other exception
+            # Maybe some custom ones should be made?
+            raise NameError('The username %s is already taken.', self.username)
+
+        with DatabaseHelper() as db:
+            db.callproc('add_user', [self.username, self.name, self.email])
+
+        return User.get_by_username(self.username).id
+
+    @classmethod
+    def get_by_id(cls, id_value):
```

Comment:
How messy would this look if it was combined into a single function? Much of the code seems to be repeated so updating the stored procedure to check the id and username in the same query might keep this simple. Something like `select * from users where id=%s or username=%s`.

Reply:
Taking a look at this now.

Reply:
I combined this into a single function to support searching either by ID or by username, pending your feedback re: [the above comment](https://github.com/CodingGrounds/INFO-3103-Final-Project/pull/17#discussion_r389727280)